### PR TITLE
virtctl: better handling of URL path for image-upload

### LINKED ISSUE #1863: virtctl image-upload fails when a trailing slash is added to the upload proxy URL
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

    $ virtctl image-upload --uploadproxy-url=https://cdi-uploadproxy-kube-system.apps.10.19.138.91.nip.io/ --pvc-name=upload-fedora-pvc4 --pvc-size=5Gi --image-path=tmp/Fedora-Cloud-Base-29-1.2.x86_64.qcow2 --insecure

Fails with:

Unexpected return value 500                                                       

Also:

    $ virtctl image-upload --uploadproxy-url=https://cdi-uploadproxy-kube-system.apps.10.19.138.91.nip.io/v1alpha/upload --pvc-name=upload-fedora-pvc4 --pvc-size=5Gi --image-path=tmp/Fedora-Cloud-Base-29-1.2.x86_64.qcow2 --insecure

Fails with:

Unexpected return value 404

**What you expected to happen**:

The URL with its path to be properly parsed and dealt with.

**How to reproduce it (as minimally and precisely as possible)**:

Execute the above commands.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):

Client Version: version.Info{GitVersion:"v0.9.6", GitCommit:"4badea07eb02458d4c64395f234fd2b77d74057e", GitTreeState:"clean", BuildDate:"2018-11-26T13:36:46Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"linux/amd64"}

Server Version: version.Info{GitVersion:"v1.3.0-13-2-g377460b", GitCommit:"4badea07eb02458d4c64395f234fd2b77d74057e", GitTreeState:"clean", BuildDate:"2018-11-26T14:36:46Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"linux/amd64"}

- Kubernetes version (use `kubectl version`): Running on OpenShift 3.11

### PR DESCRIPTION
**Pull request by tripledes**
* Use url.Parse to evaluate path in the context of the specified URL.
* Warn user if the path added on the CLI is not a valid one.

Signed-off-by: Sergi Jimenez <sjr@redhat.com>


**What this PR does / why we need it**:

Fixes #1863

**Special notes for your reviewer**:

The PR evaluates uploadProxyURI in the context of what the user specified on the command line and adds it when needed. It also removes anything that the user might have added by mistake, e.g:

    $ ./virtctl image-upload --uploadproxy-url=https://cdi-uploadproxy-kube-system.apps.10.19.138.91.nip.io/bla --pvc-name=upload-fedora-pvc12 --pvc-size=5Gi --image-path=tmp/Fedora-Cloud-Base-29-1.2.x86_64.qcow2 --insecure                                                                                                                                                                                                                                                       
    PVC kube-system/upload-fedora-pvc12 created
    Waiting for PVC upload-fedora-pvc12 upload pod to be running...
    Pod now running
    Warning, path /bla is being replaced by /v1alpha1/upload

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
virtctl image-upload better handling path on the upload proxy URL.
```

### CODE REVIEW

**Comment by tripledes** [comment]
Been checking some of the failed tests, but I don't see any failure related to my changes.

**Comment by mhenriks** [comment]
ci test please

File: pkg/virtctl/imageupload/imageupload.go [hunk 1]
```diff
@@ -202,19 +203,29 @@ func getHTTPClient(insecure bool) *http.Client {
 	return client
 }
 
-func uploadData(uploadProxyURL, token string, file *os.File, insecure bool) error {
-	url := uploadProxyURL + uploadProxyURI
+func uploadData(uploadProxyURL *url.URL, token string, file *os.File, insecure bool) error {
+	if uploadProxyURL.Path != "" && uploadProxyURL.Path != uploadProxyURI {
+		fmt.Printf("Warning, path %s is being replaced by %s\n", uploadProxyURL.Path, uploadProxyURI)
```
[the remark this answers is not in the document]
Comment (tripledes) [reply pkg/virtctl/imageupload/imageupload.go]: Originally, the code was taking the url from command line and adding `uploadProxyURI` to it, which caused issues if the user added something on the URI that wasn't right. 

Auto-discovery sounds way better indeed, I'd close this one in favor of #1876. Thanks @rmohr.

[the remark this answers is not in the document]
Comment (rmohr) [reply pkg/virtctl/imageupload/imageupload.go]: > Originally, the code was taking the url from command line and adding `uploadProxyURI` to it, which caused issues if the user added something on the URI that wasn't right.
> 

Ha, you are right.

> Auto-discovery sounds way better indeed, I'd close this one in favor of #1876. Thanks @rmohr.

Probably the best.

**Comment by tripledes** [comment]
Closing in favor of #1876

**Pull request closed without merging**